File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
int main(void)
{
    string text = get_string("Text: ");
    int textLen = strlen(text);

    int letters = 0;
    if (textLen == 0)
    {
        return 1;
    }
    int words = 1;
    int sentences = 0;

    int index;
    printf("\n");
    for (int i = 0; i < textLen; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
        else if ((text[i] >= 97 && text[i] <= 122) || (text[i] >= 65 && text[i] <= 90))
        {
            letters++;
        }
        else if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentences++;
        }
    }

    index = round(0.0588 * (letters / (double)words * 100.0) - 0.296 * (sentences / (double)words * 100.0) - 15.8);
    if (index < 16 && index >= 1)
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: sentimental-readability/readability.py
# TODO

def main():
    text = input("Text: ")
    length = len(text)

    if length == 0:
        return 1

    letters = 0
    sentences = 0
    words = 1  # count the first word

    for i in range(length):
        if text[i] == ' ':
            words += 1
        elif ((text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z')):
            letters += 1
        elif (text[i] == '!' or text[i] == '.' or text[i] == '?'):
            sentences += 1

    index = round(0.0588 * (letters / words * 100.0) - 0.296 * (sentences / words * 100.0) - 15.8)

    if (index < 16 and index >= 1):
        print(f"Grade {index}")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade 16+")


main()